### PR TITLE
refactor: Migrated superagent instrumentation into agent and no longer bundle @newrelic/superagent

### LINKED ISSUE #2128: Migrate @newrelic/superagent into repo
See https://github.com/newrelic/node-newrelic/issues/2123

Tasks:
 * deprecate the `@newrelic/superagent` package
 * Update `lib/instrumentations.js` to register a relative folder
 * Remove the external versioned tests configuration.
 * update tests to use `test/lib/agent_helper` instead of `@newrelic/test-utilities`
 * Convert git history with the previous repo.
 * Archive old repo when done.

### PR DESCRIPTION
**Pull request by jsumners-nr**
This PR resolves #2128.

### CODE REVIEW

File: lib/instrumentations.js [hunk 1]
```diff
@@ -10,39 +10,40 @@ const InstrumentationDescriptor = require('./instrumentation-descriptor')
 // Return a new copy of this array every time we're called
 module.exports = function instrumentations() {
   return {
```
Comment (jsumners-nr) [inline lib/instrumentations.js:12]: I sorted the lines in these objects. I find it a lot easier to read names in alphabetical order, and to figure out where to add a new item when the list is in alphabetical order.

File: lib/instrumentation/superagent/nr-hooks.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+module.exports = [
+  {
+    type: 'generic',
+    moduleName: 'superagent',
```
Comment (bizob2828) [inline lib/instrumentation/superagent/nr-hooks.js:11]: you don't need this. you can just have `lib/instrumentation/superagent.js`. the nr-hooks is designed when you have packages you're instrumenting that have more than once package, etc

File: lib/instrumentation/superagent/lib/http-methods.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/*
+ * Copyright 2021 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+const http = require('http')
```
Comment (bizob2828) [inline lib/instrumentation/superagent/lib/http-methods.js:7]: you could also just fold this into the `lib/instrumentation/superagent.js`

File: package.json [hunk 4]
```diff
@@ -248,6 +248,7 @@
     "rimraf": "^2.6.3",
     "should": "*",
```
Comment (bizob2828) [inline package.json:249]: you should remove `@newrelic/superagent`

**Review by bizob2828 — CHANGES_REQUESTED** [review]
A few small things, but overall looking good.

You'll need to remove superagent from [external repos](https://github.com/newrelic/node-newrelic/blob/main/test/versioned-external/external-repos.js#L31)

we will probably have to have a separate story to archive the old repo and deprecate `@newrelic/superagent`

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**